test/storage/garbage_collector_test.cpp

```diff
@@ -796,89 +812,102 @@ TEST_F(GarbageCollectorTests, InterleavedOLAP) {
     EXPECT_TRUE(StorageTestUtil::ProjectionListEqualShallow(tested.Layout(), select_tuple, update_tuple));
 
     txn_manager.Commit(txn4, TestCallbacks::EmptyCallback, nullptr);
-    // Unlink txn 4
+    // Unlink T1
     EXPECT_EQ(std::make_pair(0u, 1u), gc.PerformGarbageCollection());
 
     tested.SelectIntoBuffer(txn0, slot);
     EXPECT_FALSE(tested.select_result_);
 
     txn_manager.Commit(txn0, TestCallbacks::EmptyCallback, nullptr);
-    // Unlink read-only txn 0 and txn 7, 1  as txn 0 committed and internal transaction
-    // Deallocate txn 2, 3, 5, 6
+    // Unlink read-only T0 and U5, INSERT as T0 committed and internal transaction
+    // Deallocate  U1, U2, U3, U4
     EXPECT_EQ(std::make_pair(4u, 3u), gc.PerformGarbageCollection());
-    // Deallocate txn 7 and 1
+    // Deallocate U5 and INSERT
     EXPECT_EQ(std::make_pair(2u, 0u), gc.PerformGarbageCollection());
   }
```

Comment:
Another check here to make sure that nothing gets deallocated / unlinked after this?